zookeeper-api/src/main/java/org/apache/helix/zookeeper/datamodel/serializer/ZNRecordSerializer.java

```diff
@@ -38,6 +39,17 @@
 public class ZNRecordSerializer implements ZkSerializer {
   private static Logger logger = LoggerFactory.getLogger(ZNRecordSerializer.class);
 
+  // Reads from system property and represents whether auto compression is enabled or not.
+  // If and only if this property is set to "true", auto compression is enabled.
+  private final boolean autoCompressionEnabled =
+      Boolean.getBoolean(ZkSystemPropertyKeys.ZK_SERIALIZER_AUTO_COMPRESSION_ENABLED);
```

Comment:
The default should be ENABLED if this is for Pinot, right?

Reply:
Yes. I had a misunderstanding and with current implementation, we make it enabled by default.